### ISSUE TITLE
Issue Compiling under Fedora 27

### ISSUE THREAD

**Issue by Joykiller** [216k155/lux#74]
Below is the error compile code from console. These are the steps I took to compile under fedora..

1: ./autogen.sh
2: ./configure LDFLAGS="-Wl,-rpath=/usr/local/lib64 -L/usr/local/lib/ -L/usr/local/lib64/" CPPFLAGS="-I/usr/local/include/"
3: make

Then... the following gets this far

```
In file included from util.h:20:0,
                 from alert.cpp:14:
tinyformat.h: In static member function ‘static const char* tinyformat::detail::FormatIterator::streamStateFromFormat(std::ostream&, unsigned int&, const char*, int, int)’:
tinyformat.h:781:17: warning: this statement may fall through [-Wimplicit-fallthrough=]
         out.setf(std::ios::uppercase);
         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
tinyformat.h:782:5: note: here
     case 'x':
     ^~~~
tinyformat.h:788:17: warning: this statement may fall through [-Wimplicit-fallthrough=]
         out.setf(std::ios::uppercase);
         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
tinyformat.h:789:5: note: here
     case 'e':
     ^~~~
tinyformat.h:794:17: warning: this statement may fall through [-Wimplicit-fallthrough=]
         out.setf(std::ios::uppercase);
         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
tinyformat.h:795:5: note: here
     case 'f':
     ^~~~
tinyformat.h:799:17: warning: this statement may fall through [-Wimplicit-fallthrough=]
         out.setf(std::ios::uppercase);
         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
tinyformat.h:800:5: note: here
     case 'g':
     ^~~~
  CXX      libbitcoin_server_a-bloom.o
  CXX      libbitcoin_server_a-chain.o
  CXX      libbitcoin_server_a-checkpoints.o
  CXX      libbitcoin_server_a-init.o
In file included from spork.h:42:0,
                 from init.cpp:30:
luxsend.h:11:10: fatal error: masternode-payments.h: No such file or directory
 #include "masternode-payments.h"
          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

```

How to fix?

**Comment by Joykiller** [comment]
config log 
https://0bin.net/paste/oDjhlZRdzbbJZtlB#k+SwNhfrEcXGF9ErWU0F3r2pcYx3Sike6yjz0Qr8M6i

**Comment by Joykiller** [comment]
Downloaded the beta version ran again and its now issuing: 

`In file included from test/wallet_tests.cpp:5:0:
./wallet.h:286:10: note: candidate: bool CWallet::SelectCoinsMinConf(const string&, const CAmount&, int, int, std::vector<COutput>, std::set<std::pair<const CWalletTx*, unsigned int> >&, CAmount&) const
     bool SelectCoinsMinConf(const std::string &account, const CAmount& nTargetValue, int nConfMine, int nConfTheirs, std::vector<COutput> vCoins, std::set<std::pair<const CWalletTx*, unsigned int> >& setCoinsRet, CAmount& nValueRet) const;
          ^~~~~~~~~~~~~~~~~~
./wallet.h:286:10: note:   candidate expects 7 arguments, 6 provided
In file included from /usr/local/include/boost/test/unit_test.hpp:19:0,
                 from test/wallet_tests.cpp:13:
test/wallet_tests.cpp:281:98: error: no matching function for call to ‘CWallet::SelectCoinsMinConf(const CAmount&, int, int, std::vector<COutput>&, CoinSet&, CAmount&)’
                 BOOST_CHECK(wallet.SelectCoinsMinConf(COIN, 1, 6, vCoins, setCoinsRet2, nValueRet));
                                                                                                  ^
In file included from test/wallet_tests.cpp:5:0:
./wallet.h:286:10: note: candidate: bool CWallet::SelectCoinsMinConf(const string&, const CAmount&, int, int, std::vector<COutput>, std::set<std::pair<const CWalletTx*, unsigned int> >&, CAmount&) const
     bool SelectCoinsMinConf(const std::string &account, const CAmount& nTargetValue, int nConfMine, int nConfTheirs, std::vector<COutput> vCoins, std::set<std::pair<const CWalletTx*, unsigned int> >& setCoinsRet, CAmount& nValueRet) const;
          ^~~~~~~~~~~~~~~~~~
./wallet.h:286:10: note:   candidate expects 7 arguments, 6 provided
In file included from /usr/local/include/boost/test/unit_test.hpp:19:0,
                 from test/wallet_tests.cpp:13:
test/wallet_tests.cpp:297:101: error: no matching function for call to ‘CWallet::SelectCoinsMinConf(CAmount, int, int, std::vector<COutput>&, CoinSet&, CAmount&)’
                 BOOST_CHECK(wallet.SelectCoinsMinConf(90*CENT, 1, 6, vCoins, setCoinsRet , nValueRet));
`

**Comment by rej156** [comment]
Closing due to inactivity

**Comment by MGWalkup** [comment]
Not sure if you're still having the issue. I've found a possible work around.
The warning is merely letting you know that the switch body seems to be missing the `break;` keyword after a `case:`...

Bc some cases (pun not intended) should NOT have the break, you can silence the warning (the block below as an example). 

[The Redhat Developer Blog has a good explanation](https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7/)

TL;DR:
It depends on the version of C++ you're using:


switch (cond)
 {
case 1:
   bar (1);
   __attribute__ ((fallthrough)); // C and C++03
 case 2:
   bar (2);
   [[gnu::fallthrough]]; // C++11 and C++14
 case 3:
   bar (3);
   [[fallthrough]]; // C++17 and above
 /* ... */
 }
